rtCommon/bidsIncremental.py

```diff
@@ -0,0 +1,166 @@
+    # which might be different than the expected behavior
+    removalRegex = re.compile('[^a-zA-z]')
+    ignoredTags = ['Pixel Data']
+
+    for elem in dicomImg:
+        if elem.name in ignoredTags:
+            continue
+
+        cleanedKey = removalRegex.sub("", elem.name)
+        # in DICOM, public tags have even group numbers and private tags are odd
+        # http://dicom.nema.org/dicom/2013/output/chtml/part05/chapter_7.html
+        if elem.tag.group % 2 == 0:
+            publicMeta[cleanedKey] = str(elem.value)
+        else:
+            privateMeta[cleanedKey] = str(elem.value)
+
+    return (publicMeta, privateMeta)
+
+
+def dicomToBidsinc(dicomImg: pydicom.dataset.Dataset) -> BidsIncremental:
+    # TODO(spolcyn): Do this all in memory -- dicom2nifti is promising
+    # Put extra metadata in sidecar JSON file
+    niftiImage = convertDicomImgToNifti(dicomImg)
+    publicMeta, privateMeta = getMetadata(dicomImg)
+
+    publicMeta.update(privateMeta)  # combine metadata dictionaries
+    return BidsIncremental(niftiImage, publicMeta)
+
+def appendBidsinc(incremental: BidsIncremental, archive: BidsArchive) -> None:
+    """
+    Appends the provided BIDS Incremental imaging and metadata to the provided
+    BIDS archive.
+    Args:
+        incremental: BIDS Incremental file containing image data and metadata
+        archive: BIDS Archive file to append image data and metadata to
+    Returns:
+        None
+    """
+    """
+    General Strategy:
+    1) Find the path of the NIfTI to append to
+    2) Verify the path and NIfTI's existence
+    3) Load in the NIfTI and get its data
+    4) Use numpy to append the BIDS-I data to the archive data
+    5) Update the archive NIfTI files
+    6) Update the header, if necessary
+    7) Update the metadata, if necessary
+    8) Write the NIfTI back to disk (somewhat inefficient b/c must write the
+    whole NIfTI instead of just the modified parts)
+    """
+    # 1) Find Nifti path
+    imgDirPath = incremental.makeDataDirPath()
+    imgPath = os.path.join(imgDirPath, incremental.getImageFileName())
+    imgDirPath = os.path.join(imgDirPath, "")
+    logger.debug("Image dir path: %s", imgDirPath)
+    logger.debug("Image file path: %s", imgPath)
+
+    # 2) Verify we have a valid way to append the image to the archive. 3 cases:
+    # 2.1) Image already exists within archive, append this Nifti to that Nifti
+    # 2.2) Image doesn't exist in archive, but rest of the path is valid for the
+    # archive; create new Nifti file within the archive
+    # 2.3) Neither image nor path is valid for provided archive; fail append
+    if archive.pathExists(imgPath):
+        # do append
+        logger.debug("Image exists in archive, appending")
+    elif archive.pathExists(imgDirPath):
+        # create new image
+        logger.debug("Image doesn't exist in archive, creating")
+    else:
+        # fail
+        logger.debug("No valid archive path for image, can't insert or append")
```

Comment:
Maybe have an option to create the directory if it doesn't exist, like makeDirs=False by default.